src/explore-education-statistics-common/src/components/form/FormCheckboxSearchGroup.tsx

```diff
@@ -20,13 +20,17 @@ export interface FormCheckboxSearchGroupProps extends FormCheckboxGroupProps {
 
 const FormCheckboxSearchGroup = ({
   searchLabel,
+  searchOnly = false,
   ...props
 }: FormCheckboxSearchGroupProps) => {
   const { isMounted } = useMounted();
 
   const [searchTerm, setSearchTerm] = useState('');
 
-  if (!isMounted) {
+  // searchOnly is used when we expect a very large number
+  // of options which would cause very slow rendering
+  // if they were all rendered.
+  if (!isMounted && !searchOnly) {
```

Comment:
So do we not have any tests that would potentially catch this issue should it arise in the future?  E.g. a test that sets searchOnly to true and asserts that there are no checkboxes initially rendered?

Reply:
We've already got tests for this, but they wouldn't pick up this issue as the tests run with JS so the component would be mounted almost instantly and the checkboxes would never be visible.